src/pkg/egress/syslog/https_batch.go

```diff
@@ -0,0 +1,116 @@
+package syslog
+
+import (
+	"bytes"
+	"crypto/tls"
+	"time"
+
+	"code.cloudfoundry.org/go-loggregator/v9/rpc/loggregator_v2"
+	metrics "code.cloudfoundry.org/go-metric-registry"
+	"code.cloudfoundry.org/loggregator-agent-release/src/pkg/egress"
+)
+
+const BATCHSIZE = 256 * 1024
+
+type HTTPSBatchWriter struct {
+	HTTPSWriter
+	msgBatch     bytes.Buffer
+	batchSize    int
+	sendInterval time.Duration
+	sendTimer    *TriggerTimer
+	egrMsgCount  float64
+}
+
+func NewHTTPSBatchWriter(
+	binding *URLBinding,
+	netConf NetworkTimeoutConfig,
+	tlsConf *tls.Config,
+	egressMetric metrics.Counter,
+	c *Converter,
+) egress.WriteCloser {
+	client := httpClient(netConf, tlsConf)
+	binding.URL.Scheme = "https" // reset the scheme for usage to a valid http scheme
```

Comment:
What's the purpose of changing the scheme here?

Reply:
the scheme is used to differentiate between the different endpoints (https and https-batched)
If i do not change it back to https for sends, the queried url will be https-batched://... , which is just not working then.

Reply:
I will check for the metrics  being changed here.

Reply:
Did the metrics change?

Reply:
need to test this one again, will do so anyway today.